node.go

```diff
@@ -439,6 +439,85 @@ func (node *Node) traverseInRange(t *ImmutableTree, start, end []byte, ascending
 	return stop
 }
 
+// This method doesn't hold any reference to loaded node, its cb's responsibility to decide whether hold Node reference
+// If cb (like usage in state sync) don't need hold a reference to loaded Node, the memory footprint will be near to zero
+func (node *Node) traverseInRangeDiscardNode(t *ImmutableTree, start, end []byte, ascending bool, inclusive bool, depth uint8, cb func(*Node, uint8) bool) bool {
```

Comment:
Total agree with your logic.  What if we can reuse the origin code :
```
func (node *Node) traverseInRangeDiscardNode(t *ImmutableTree, start, end []byte, ascending bool, inclusive bool, depth uint8, cb func(*Node, uint8) bool) bool {
defer func(){
		node.leftNode = nil
		node.rightNode = nil
}()
     return node.traverseInRange(t, nil, nil, ascending, false, 0, func(node *Node, depth uint8) bool {
		return cb(node)
	})
}
```

Reply:
not cheap to use defer, especially when there is a recursive call